schema.go

```diff
@@ -622,6 +708,12 @@ func (s *RecordSchema) String() string {
 		fields = fields[:len(fields)-1]
 	}
 
+	if s.logical != nil {
+		return fmt.Sprintf("{\"name\":\"%s\", \"type\":\"%s\", \"fields\":[%s]\", %s}",
```

Comment:
`fmt.Sprintf` is slow and includes many allocations. Given this function is called often, this is a performance bottleneck.